Referee.java

```diff
@@ -1,27 +1,45 @@
 // This is the class that will make the animals compete. All the logic of how the battle is calculated is done from this class
 
 import java.util.ArrayList;
+import java.util.Arrays;
 
 public class Referee {
 
   EvaluationCategory fightingCategory;
   EvaluationCategory movementCategory;
-  EvaluationCategory[] competitionCategories = new EvaluationCategory[2];
+  EvaluationCategory survivalCategory;
+  EvaluationCategory[] competitionCategories = new EvaluationCategory[3];
```

Comment:
You could move the initialization of `competitionCategories` to the constructor so you don't have to know it's size here:
```java
this.competitionCategories = new EvaluationCategory[]{this.fightingCategory, this.movementCategory, this.survivalCategory};
```

Reply:
Generally, the less knowledge something needs, the better; we always strive for minimal dependencies.